lib/http-proxy/passes/web-incoming.js

```diff
@@ -162,16 +162,22 @@ module.exports = {
 
     proxyReq.on('response', function(proxyRes) {
       if(server) { server.emit('proxyRes', proxyRes, req, res); }
-      for(var i=0; i < web_o.length; i++) {
-        if(web_o[i](req, res, proxyRes, options)) { break; }
+      if (!res.headersSent) {
+        for(var i=0; i < web_o.length; i++) {
+          if(web_o[i](req, res, proxyRes, options)) { break; }
```

Comment:
Hi @frogmella,

I think that your your commit is ok and we should avoid the writeHead function. But if the headers are already sent, does not makes sense to call ```res.statusCode = proxyRes.statusCode;```. The response status code and headers was already sent to client, so you can not modify it anymore.